### PR TITLE
[webview_flutter] Migrate webview_flutter_wkwebview to analysis_options.yaml

### PR DESCRIPTION
**Pull request by mvanbeusekom**
Migrates the webview_flutter_wkwebview package from using `analysis_options_legacy.yaml` to the more strict `analysis_options.yaml`.

No version change: only updates to the more strict analysis_options.yaml and doesn't make any functional changes that would require a new version to be published. I added a line to the already existing "NEXT" section of the CHANGELOG.md.

Related to flutter/flutter#76229

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_wkwebview/example/integration_test/webview_flutter_test.dart [hunk 1]
```diff
@@ -845,7 +845,8 @@ void main() {
   });
 
   group('NavigationDelegate', () {
-    final String blankPage = "<!DOCTYPE html><head></head><body></body></html>";
+    const String blankPage =
+        '<!DOCTYPE html><head></head><body></body></html>;';
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_wkwebview/example/integration_test/webview_flutter_test.dart:849]: You added a character to the end of the string here

**Review by stuartmorgan — APPROVED** [review]
Nice, thanks! Please make sure to remove this package from the allowed overrides in the script/config YAML file.

LGTM with that and the one issue below fixed.

File: packages/webview_flutter/webview_flutter_wkwebview/example/integration_test/webview_flutter_test.dart (hunk 1 above)
Comment (mvanbeusekom) [reply packages/webview_flutter/webview_flutter_wkwebview/example/integration_test/webview_flutter_test.dart]: Oops that was indeed not my intention. Fixed.

**Comment by mvanbeusekom** [comment]
> Please make sure to remove this package from the allowed overrides in the script/config YAML file.

I did check the `scripts/config/custom_analysis.yaml` file but it only contains an entry for the "webview_flutter" app-facing package. Since the "webview_flutter_platform_interface" and "webview_flutter" packages still reference the `analysis_options_legacy.yaml` file I haven't removed this entry. 

My plan was to also update the above packages and when I do the app facing package also remove the line from the configuration. Would this make sense?

**Comment by stuartmorgan** [comment]
It sounds like it wasn't updated when the legacy file was pushed from the top level into the individual packages; you should be able to replace the general entry with individual `webview_flutter/webview_flutter_foo` entries for things still using legacy files.

**Comment by stuartmorgan** [comment]
(But it's also not a big deal to wait until they are all done; LGTM either way.)

**Comment by mvanbeusekom** [comment]
Thanks @stuartmorgan, I will do both ;). Just updated the configuration in the custom_analysis.yaml file.

**Pull request merged**